### PR TITLE
feat:[MTL] Enable S0ix feature on MTL platform

### PR DESCRIPTION
**Pull request by kokweich**
Enable S0ix on MTL platform
 - Sync ASL and NVS value with MTL BIOS
 - Added workaround for RP01 in pep.asl to enter s0ix

### CODE REVIEW

File: Platform/MeteorlakeBoardPkg/AcpiTables/Dsdt/MtlPchNvs.asl [hunk 1]
```diff
@@ -0,0 +1,234 @@
+//
+// Automatically generated by GenNvs ver 2.4.6
+// Please DO NOT modify !!!
+//
+
+/**@file
+
+@copyright
+  INTEL CONFIDENTIAL
+  Copyright 2021 - 2022 Intel Corporation.
```
Comment (ywho79) [inline Platform/MeteorlakeBoardPkg/AcpiTables/Dsdt/MtlPchNvs.asl:10]: 2023

File: Platform/MeteorlakeBoardPkg/Library/Stage2BoardInitLib/Stage2BoardInitLib.c [hunk 2]
```diff
@@ -1131,29 +1111,30 @@ PlatformUpdateAcpiTable (
           Address = MSR_PC10_RCNTR;
           ResidencyCounterFrequency = 0; //Counter runs at TSC frequency
           UpdLpiStat((ACPI_LOW_POWER_IDLE_TABLE *)Table, 0, Flags, Address, ResidencyCounterFrequency);
-          if (IsPchP () || IsPchN ()) {
+          // if (IsPchP () || IsPchN ()) {
```
Comment (gdong1) [inline Platform/MeteorlakeBoardPkg/Library/Stage2BoardInitLib/Stage2BoardInitLib.c:1114]: Please remove commented out code, including line 1122 ~ 1130

File: Platform/MeteorlakeBoardPkg/Library/Stage2BoardInitLib/LowPowerSupport.c [hunk 3]
```diff
@@ -258,47 +270,45 @@ UINT64 GetLowPowerS0IdleConstraint(VOID)
...
-                                ((UINT64)PepConfigData->PepPcieDg                                                  << 43) | // Bit[43]    - En/Dis DG on x8 PEG port (PEG1) //SG- if PcdAdlLpSupport = 1
-                                ((UINT64)(PepUfs[0] && PepConfigData->PepUfs0)                                     << 44) | // Bit[44]    - En/Dis UFS0
-                                ((UINT64)(PepUfs[0] && PepConfigData->PepUfs1)                                     << 45) | // Bit[45]    - En/Dis UFS1
-                                ((UINT64)(PepSerialIoI2c[7] && PepConfigData->PepI2c7)                             << 46);  // Bit[46]    - En/Dis I2C7
+    LowPowerS0IdleConstraint = ((PepSerialIoUart[0] && PepConfigData->PepUart)                         << 0) | // En/Dis BIT[0] UART0
+                                                            ((PepSerialIoUart[1] && PepConfigData->PepUart)                         << 1) | // En/Dis BIT[1] UART1
+                                                            ((PepSerialIoUart[2] && PepConfigData->PepUart)                         << 2) | // En/Dis BIT[2] UART2
```
Comment (gdong1) [inline Platform/MeteorlakeBoardPkg/Library/Stage2BoardInitLib/LowPowerSupport.c:275]: Please fix the alignment.

File: Platform/MeteorlakeBoardPkg/Library/Stage2BoardInitLib/LowPowerSupport.c [hunk 4]
```diff
@@ -170,9 +170,9 @@ UINT64 GetLowPowerS0IdleConstraint(VOID)
       PepConfigData->PepVmd = 0;
     }
 
-    if (PciRead16 (PCI_SEGMENT_LIB_ADDRESS (0, SA_PEG_BUS_NUM, SA_PEG1_DEV_NUM, SA_PEG0_FUN_NUM, PCI_VENDOR_ID_OFFSET)) == 0xFFFF) {
-      PepConfigData->PepPcieDg = 0;
-    }
+    // if (PciRead16 (PCI_SEGMENT_LIB_ADDRESS (0, SA_PEG_BUS_NUM, SA_PEG1_DEV_NUM, SA_PEG0_FUN_NUM, PCI_VENDOR_ID_OFFSET)) == 0xFFFF) {
+    //   PepConfigData->PepPcieDg = 0;
+    // }
```
Comment (gdong1) [inline Platform/MeteorlakeBoardPkg/Library/Stage2BoardInitLib/LowPowerSupport.c:175]: Remove it if it is not required.

File: Platform/MeteorlakeBoardPkg/Library/FspsUpdUpdateLib/FspsUpdUpdateLib.c [hunk 5]
```diff
@@ -835,4 +825,23 @@ UpdateFspConfig (
 
   // EndOfPost Upd
   FspsUpd->FspsConfig.EndOfPostMessage = 1;
+    if (FeaturesCfgData != NULL) {
```
Comment (gdong1) [inline Platform/MeteorlakeBoardPkg/Library/FspsUpdUpdateLib/FspsUpdUpdateLib.c:828]: Fix the alignment.

File: Platform/MeteorlakeBoardPkg/AcpiTables/Ssdt/MtlPSCrbRtd3.asl [hunk 6]
```diff
@@ -0,0 +1,309 @@
...
+// PCIe root ports - END
+
+//
+// VMD - START
+//
+// #if (FixedPcdGetBool (PcdVmdEnable) == 1)
+//   Include ("Rtd3Vmd.asl")
+// #endif
+//
+// VMD - END
```
Comment (gdong1) [inline Platform/MeteorlakeBoardPkg/AcpiTables/Ssdt/MtlPSCrbRtd3.asl:240]: Remove VMD above

File: Platform/MeteorlakeBoardPkg/AcpiTables/Dsdt/MtlPchNvs.asl [hunk 7]
```diff
@@ -0,0 +1,234 @@
...
+  express and approved by Intel in writing.
+
+  Unless otherwise agreed by Intel in writing, you may not remove or alter
+  this notice or any other notice embedded in Materials by Intel or
+  Intel's suppliers or licensors in any way.
+
+  This file contains an 'Intel Peripheral Driver' and is uniquely identified as
+  "Intel Reference Module" and is licensed for Intel CPUs and chipsets under
+  the terms of your license agreement with Intel or your vendor. This file may
+  be modified by the user, subject to additional terms of the license agreement.
```
Comment (gdong1) [inline Platform/MeteorlakeBoardPkg/AcpiTables/Dsdt/MtlPchNvs.asl:35]: Please use SPDX-License-Identifier: BSD-2-Clause-Patent

File: Platform/MeteorlakeBoardPkg/AcpiTables/Dsdt/MtlPchNvs.asl (hunk 1 above)
Comment (kokweich) [reply Platform/MeteorlakeBoardPkg/AcpiTables/Dsdt/MtlPchNvs.asl]: Resolved in latest push

File: Platform/MeteorlakeBoardPkg/Library/Stage2BoardInitLib/Stage2BoardInitLib.c [hunk 8]
```diff
@@ -1805,14 +1768,41 @@ PlatformUpdateAcpiGnvs (
...
+
   PlatformNvs->WwanWakeGpio = GPIOV2_MTL_SOC_M_GPP_F21;
   PlatformNvs->WwanRootPortNumber = 0x7;
   PlatformNvs->WwanFwFlashDevice = 0;
-  PlatformNvs->XdciFnEnable = 0;
-
-  PlatformNvs->LidSwitchWakeGpio = GPIOV2_MTL_SOC_M_GPP_V2;
-  DEBUG ((DEBUG_INFO, "PlatformNvs->LidSwitchWakeGpio: 0x%x\n", PlatformNvs->LidSwitchWakeGpio));
+  PlatformNvs->XdciFnEnable = 1;
+  // DEBUG ((DEBUG_INFO, "WwanWakeGpio = 0x%x\n", PlatformNvs->WwanWakeGpio));
```
[the remark this answers is not in the document]
Comment (kokweich) [reply Platform/MeteorlakeBoardPkg/Library/Stage2BoardInitLib/Stage2BoardInitLib.c]: Fixed in latest push

File: Platform/MeteorlakeBoardPkg/Library/Stage2BoardInitLib/Stage2BoardInitLib.c (hunk 2 above)
Comment (kokweich) [reply Platform/MeteorlakeBoardPkg/Library/Stage2BoardInitLib/Stage2BoardInitLib.c]: fixed in latest push

File: Platform/MeteorlakeBoardPkg/Library/Stage2BoardInitLib/LowPowerSupport.c (hunk 3 above)
Comment (kokweich) [reply Platform/MeteorlakeBoardPkg/Library/Stage2BoardInitLib/LowPowerSupport.c]: fixed in latest push

File: Platform/MeteorlakeBoardPkg/Library/Stage2BoardInitLib/LowPowerSupport.c (hunk 4 above)
Comment (kokweich) [reply Platform/MeteorlakeBoardPkg/Library/Stage2BoardInitLib/LowPowerSupport.c]: removed commented out lines in latest push

File: Platform/MeteorlakeBoardPkg/AcpiTables/Ssdt/MtlPSCrbRtd3.asl (hunk 6 above)
Comment (kokweich) [reply Platform/MeteorlakeBoardPkg/AcpiTables/Ssdt/MtlPSCrbRtd3.asl]: Removed in latest push

File: Platform/MeteorlakeBoardPkg/AcpiTables/Dsdt/MtlPchNvs.asl (hunk 7 above)
Comment (kokweich) [reply Platform/MeteorlakeBoardPkg/AcpiTables/Dsdt/MtlPchNvs.asl]: Updated license in latest push

**Review by ywho79 — APPROVED** [review]
Looks good to me

**Review by gdong1 — APPROVED** [review]

**Pull request merged**